flow/core/kernel/pedestrian/traci.py

```diff
@@ -0,0 +1,28 @@
+
+from flow.core.kernel.pedestrian import KernelPedestrian
+import traci.constants as tc
+import numpy as np
+
+class TraCIPedestrian(KernelPedestrian):
+
+    def __init__(self,
+            master_kernel):
+
+        KernelPedestrian.__init__(self, master_kernel)
+
+        self.__ids = [] # ids of all pedestrians
+
+    def get_speed(self, ped_id, error=-1001):
+        if isinstance(ped_id, (list, np.ndarray)):
+            return [self.get_speed(pedID, error) for pedID in ped_id]
+        return self.kernel_api.person.getSpeed(ped_id)
+
+    def get_position(self, ped_id, error=-1001):
+        if isinstance(ped_id, (list, np.ndarray)):
+            return [self.get_position(pedID, error) for pedID in ped_id]
+        return self.kernel_api.person.getPosition(ped_id)
+
+    def get_edge(self, ped_id, error=-1001):
+        if isinstance(ped_id, (list, np.ndarray)):
+            return [self.get_edge(pedID, error) for pedID in ped_id]
+        return self.kernel_api.person.getRoadID(ped_id)
```

Comment:
Think we may want to do something similar to how its done in the TraciVehicle where at each timestep we log all the info to a dict and then access it from a dict later. It's faster since we can use subscriptions to batch the calls to TraCI so we're not accessing it constantly.

Reply:
fixed